discovery-infra/test_infra/kubeapi_utils.py

```diff
@@ -91,3 +96,24 @@ def get_hostname_for_agent(agent, nodes_details):
             if interface['macAddress'].lower() == mac_address:
                 return node_metadata['name']
 
+
+def delete_kube_api_resources_for_namespace(kube_api_client, name, namespace):
+    # any resource that created within a KubeAPIContext will be deleted on
+    # context exit.
+    with KubeAPIContext(kube_api_client):
+        cluster_deployment = ClusterDeployment(
+            kube_api_client=kube_api_client,
+            name=name,
+            namespace=namespace,
+        )
+        cluster_deployment.list_agents()
+        Secret(
+            kube_api_client=kube_api_client,
+            name=name,
```

Comment:
Is it safe to assume the sercet and clusterdeployment have the same name?

Reply:
nope, not safe to assume that, thanks